fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -343,14 +358,22 @@ public void close() {
     /**
      * Run after a single transactional call within runAsync.
      */
-    private <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception) {
-        if (exception == null) {
-            successCount++;
-            if (successCount >= increaseLimitAfter && limit < maxLimit) {
-                increaseLimit();
+    @VisibleForTesting
+    <R> Pair<R, Throwable> runAsyncPostTransaction(R result, Throwable exception, AtomicLong recordsScanned) {
```

Comment:
Can this have nullability annotations, or does that not play nicely with the type system in the lambda expressions?

Reply:
Seems to be just fine with the annotations.